platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/ConsensusRoundHandler.java

```diff
@@ -456,43 +254,48 @@ private void updatePlatformState(final ConsensusRound round) {
     }
 
     /**
-     * Update the running event hash in the platform state.
+     * Update the running hash of the consensus events in the platform state
+     * <p>
+     * This method is called after the consensus round has been applied to the state, to minimize the chance that
+     * we have to wait for the running hash to finish being calculated.
+     *
+     * @param round the consensus round
+     * @throws InterruptedException if this thread is interrupted
      */
-    private void updateRunningEventHash() throws InterruptedException {
+    private void updatePlatformStateRunningHash(@NonNull final ConsensusRound round) throws InterruptedException {
         final PlatformState platformState =
                 swirldStateManager.getConsensusState().getPlatformState();
-        final Hash runningHash = eventsConsRunningHash.getFutureHash().getAndRethrow();
+
+        // Update the running hash object. If there are no events, the running hash does not change.
+        // Future work: this is a redundant check, since empty rounds are currently ignored entirely. The check is here
+        // anyway, for when that changes in the future.
+        if (!round.isEmpty()) {
+            consensusEventsRunningHash = round.getConsensusEvents().getLast().getRunningHash();
+        }
+
+        final Hash runningHash = consensusEventsRunningHash.getFutureHash().getAndRethrow();
         platformState.setRunningEventHash(runningHash);
     }
 
+    /**
+     * Create a signed state
+     *
+     * @throws InterruptedException if this thread is interrupted
+     */
     private void createSignedState() throws InterruptedException {
-        final CycleTimingStat ssTimingStat = consensusHandlingMetrics.getNewSignedStateCycleStat();
-        ssTimingStat.startCycle();
+        if (freezeRoundReceived) {
+            // Let the swirld state manager know we are about to write the saved state for the freeze period
+            swirldStateManager.savedStateInFreezePeriod();
+        }
 
+        handlerMetrics.setPhase(GETTING_STATE_TO_SIGN);
         // create a new signed state, sign it, and send out a new transaction with the signature
```

Comment:
out of date comment - this method does not sign the state or send out a transaction

Reply:
Fixed